openwisp_monitoring/device/api/serializers.py

```diff
@@ -20,6 +26,79 @@ class Meta:
         model = DeviceMonitoring
 
 
+class WifiClientSerializer(FilterSerializerByOrgManaged, ValidatedModelSerializer):
+    class Meta:
+        model = WifiClient
+        fields = [
+            'mac_address',
+            'vendor',
+            'ht',
+            'vht',
+            'wmm',
+            'wds',
+            'wps',
+            'modified',
+            'created',
+        ]
+        read_only_fields = (
+            'created',
+            'modified',
+        )
+
+
+class WifiSessionCreateUpdateSerializer(
+    FilterSerializerByOrgManaged, ValidatedModelSerializer
+):
+    organization_lookup = 'wifisession__device__organization__in'
```

Comment:
This is causing the error of `MultipleObjectsRetuned`. This filters the Device object with following query: 
```
Device.objects.filter(wifisession_device_organization__in=['users-organization'])
```
This creates a join between `Device` and `WifiSession` and since there are multiple WifiSessions, it returns two device objects in the query: 

```
<QuerySet [<Device: default.test.device>, <Device: default.test.device>]>
```

This is the SQL query which is executed 
```
SELECT "config_device"."id", "config_device"."created", "config_device"."modified", "config_device"."organization_id", "config_device"."name", "config_device"."mac_address", "config_device"."key", "config_device"."model", "config_device"."os", "config_device"."system", "config_device"."notes", "config_device"."group_id", "config_device"."last_ip", "config_device"."management_ip", "config_device"."hardware_id" FROM "config_device" INNER JOIN "device_monitoring_wifisession" ON ("config_device"."id" = "device_monitoring_wifisession"."device_id") INNER JOIN "config_device" T3 ON ("device_monitoring_wifisession"."device_id" = T3."id") WHERE T3."organization_id" IN (18e9ef4bb6d64f768a4cf17de94a28b1)
```

I have opened an issue for this in openwisp-users https://github.com/openwisp/openwisp-users/issues/319

Reply:
@nemesisdesign what do you suggest to do here? Should we complete this PR without the filtering for device and wifi-client fields and open an issue for it?

Reply:
Can you tell me the steps to replicate this issue please?

Reply:
@nemesisdesign Aryaman wrote a test for this in https://github.com/openwisp/openwisp-monitoring/pull/382/commits/aa9a0f5c18c341700b379610f9e1b62b14910063

Reply:
Hey @nemesisdesign, what you do think about https://github.com/openwisp/openwisp-monitoring/pull/382#discussion_r900104443?

Reply:
I think the problem is first of all a design issue.

Let me explain the initial design.

We store the clients to avoid repeating the same information when a single client performs multiple sessions, but since a client could perform WiFi sessions in multiple organizations, it doesn't have the organization field, otherwise we would have to duplicate the client information in each organization and the goal of avoiding the repetition would not be met.

If we analyze the statement above more closely, we can figure out that the `WiFiClient` model is an implementation detail of the WiFi Sessions feature, which we hide in the admin interface, therefore I am thinking that probably my initial request to add an API endpoint for it was wrong in the first place.

Probably we have to expose only the session list information with the appropriate list filters (as in the admin) I am not sure allowing to create a session even has any use case right now.

@pandafy @Aryamanz29 what do you think?